test-suite/js/src/test/scala/org/scalajs/testsuite/library/ObjectTest.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.library
+
+import org.junit.Assert._
+import org.junit.Assume._
+import org.junit.Test
+import org.scalajs.testsuite.utils.Platform
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.{ JSBracketAccess, JSName }
+
+class ObjectTest {
+  import ObjectTest._
+
+  @Test def getOwnPropertyDescriptors(): Unit = {
+    assumeTrue(
+      "Assuming newer methods existence, which are not honored by Rhino",
+      Platform.executingInNodeJS && !Platform.executingInRhino)
+    val obj1 = new js.Object {
+      val property1 = 42
+    }
+    val propertyDescriptors = js.Object.getOwnPropertyDescriptors(obj1)
+    val property = propertyDescriptors("property1")
+    assertTrue(property.writable)
+    assertEquals(property.value, 42)
+  }
+
+  private val symA: js.Symbol = js.Symbol("a")
+  private val symB: js.Symbol = js.Symbol.forKey("b")
+
+  @Test def getOwnPropertySymbols(): Unit = {
+    assumeTrue(
+      "Assuming newer methods existence, which are not honored by Rhino",
+      Platform.executingInNodeJS && !Platform.executingInRhino)
+    val obj = (new js.Object()).asInstanceOf[ObjectCreator]
+    obj(symA) = "localSymbol"
+    obj(symB) = "globalSymbol"
+    val objectSymbols = js.Object.getOwnPropertySymbols(obj)
+    assertArrayEquals(Array[AnyRef](symA, symB), objectSymbols.toArray[AnyRef])
+  }
+
+  @Test def is(): Unit = {
+    assumeTrue(
+      "Assuming newer methods existence, which are not honored by Rhino",
+      Platform.executingInNodeJS && !Platform.executingInRhino)
+    val a = new js.Object()
+    assertTrue(js.Object.is(a, a))
+    assertTrue(js.Object.is(Double.NaN, Double.NaN))
+    assertTrue(js.Object.is(0.0, +0.0))
+    assertTrue(js.Object.is(-0.0, -0.0))
+    assertTrue(js.Object.is("foo", "foo"))
+
+    val b = new js.Object()
+    assertFalse(js.Object.is(a, b))
+    assertFalse(js.Object.is(-0.0, +0.0))
+  }
+
+  @Test def entries_from_object(): Unit = {
+    assumeTrue(
+      "Assuming newer methods existence, which are not honored by Rhino",
+      Platform.executingInNodeJS && !Platform.executingInRhino)
+    val obj = new js.Object {
+      val a = 42
+      val b = "foo"
+    }
+    val entries = js.Object.entries(obj)
+    assertEquals(2, entries.length)
+
+    val js.Tuple2(key1, value1) = entries(0)
+    assertEquals("a", key1)
+    val value1IsInt: Int = value1.asInstanceOf[Int]
```

Comment:
This makes no sense for the overload with `js.Object`. Casting to `Int` and then checking that it can be assigned to an `Int` is a no-op.

For this overload, just use `value1` in `assertEquals` below.

The `value1IsInt` trick is good in the overload with a `js.Dictionary`, which you did in the test below.